_episodes/03-running-containers.md

```diff
@@ -0,0 +1,234 @@
+
+| 1. Starts a Running Container | 2. Performs Default Action | 3. Shuts Down the Container
+| --------------------|-----------------|----------------|
+| starts a running container, based on the image. Think of this as the "alive" or
+"inflated" version of the container -- it's actually doing something | If the container has a default action set, it will perform that default action. This could be as simple as printing a message (as above) or running a whole analysis pipeline! | Once the default action is complete, the container stops running (or exits). The image
+is still there, but nothing is actively running. |
+
+The `hello-world` container is set up to run an action by default -
+namely to print this message.
+
+> ## Using `docker run` to get the image
+>
+> Technically, we could have skipped the `docker pull` step; if you use the `docker run`
+> command and you don't already have a copy of the Docker image, Docker will
+> automatically pull the image first and then run it.
+{: .callout}
+
+## Running a container with a chosen command
+
+But what if we wanted to do something different with the container? The output
+just gave us a suggestion of what to do -- let's use a different Docker image
+to explore what else we can do with the `docker run` command. The suggestion above
+is to use `ubuntu`, but we're going to run a different type of Linux, `alpine`
+instead because it's quicker to download.
+
+> ## Run the Alpine Docker container
+>
+> Try downloading and running the `alpine` Docker container. You can do it in
+> two steps, or one. What are they?
+{: .callout}
+
+What happened when you ran the Ubuntu Docker container?
+
+~~~
+$ docker run alpine
+~~~
+{: .language-bash}
+
+Probably nothing! That's because this particular container is designed for you to
+provide commands yourself. Try running this instead:
+
+~~~
+$ docker run alpine cat /proc/version
+~~~
+{: .language-bash}
+
+You should see the output of the `cat /proc/version` command, which prints out
+the version of Linux that this container is using.
+
+> ## Exercise
+> Can you run the container and make it print a "hello world" message?
+>
+> Give it a try before checking the solution.
+>
+> > ## Solution
+> >
+> > Use the same command as above, but with the `echo` command to print a message.
+> > ~~~
+> > $ docker run alpine echo 'Hello World'
+> > ~~~
+> > {: .language-bash}
+> {: .solution}
+{: .challenge}
+
+So here, we see another option -- we can provide commands at the end of the `docker run`
+command and they will execute inside the running container.
+
+## Running containers interactively
+
+In all the examples above, Docker has started the container, run a command, and then
+immediately shut down the container. But what if we wanted to keep the container
+running so we could log into it and test drive more commands? The way to
+do this is by adding the interactive flag `-it` to the `docker run` command and
```

Comment:
This might be too in the weeds but it is actually two flags `-i` for interactive and then `-t` for "Allocate a pseudo-TTY".  I had to look up what the `-t` did myself, some info [here](https://stackoverflow.com/questions/30137135/confused-about-docker-t-option-to-allocate-a-pseudo-tty)

Reply:
I'll maybe make a callout?